io/emoncms/88-emoncms.js

```diff
@@ -75,9 +73,50 @@ module.exports = function(RED) {
                 // node.error(e);
                 msg.rc = 503;
                 msg.payload = e;
-                node.send(msg);
             });
         });
     }
     RED.nodes.registerType("emoncms",Emoncms);
+ 
+    function Emoncmsin(n) {
+        RED.nodes.createNode(this,n);
+        this.emonServer = n.emonServer;
+        var sc = RED.nodes.getNode(this.emonServer);
+
+        this.baseurl = sc.server;
+        this.apikey = sc.credentials.apikey;
+
+        this.feedid = n.feedid
+        var node = this;
+        var http;
+        if (this.baseurl.substring(0,5) === "https") { http = require("https"); }
+        else { http = require("http"); }
+        this.on("input", function(msg) {
+            this.url = this.baseurl + '/feed/value.json';
+            this.url += '&apikey='+this.apikey;
+            var feedid = this.feedid || msg.feedid;
+            if (feedid !== "") {
+                this.url += '&id=' + feedid;
+            }
+            http.get(this.url, function(res) {
+                msg.rc = res.statusCode;
+                msg.payload = "";
+                if ((msg.rc != 200) && (msg.rc != 404)) {
+                    node.send(JSON.parse(msg));
```

Comment:
This needs moving down to line 112, inside the on(end) handler to replace the node.send call already there.